internal/api/cluster.go

```diff
@@ -362,8 +362,9 @@ func handleUpgradeKubernetes(c *Context, w http.ResponseWriter, r *http.Request)
 		w.WriteHeader(http.StatusInternalServerError)
 		return
 	}
-	if kopsMetadata.Version != version {
-		kopsMetadata.Version = version
+	if kopsMetadata.Version != upgradeClusterRequest.Version || kopsMetadata.AMI != upgradeClusterRequest.KopsAMI {
```

Comment:
Are we stylistically opposed to putting a newline before or after the `||` in compound boolean statements like this? Personally I think they're a little easier one per line but 0/5 ignore me if you don't like the suggestion :)

Reply:
This actually reminded me that I had originally intended to use patching logic and have the `Apply` method take care of reporting if the metadata was updated. New commit has this logic!